swagger.yaml

```diff
@@ -751,7 +751,99 @@ paths:
                 items:
                   $ref: "#/components/schemas/Error"
 
-  "/vacationsRequests":
+  "/severa/projects/{severaProjectGuid}/phaseswithhierarchy":
+    get:
+      operationId: getPhasesBySeveraProjectId
+      summary: Get all the phases for project
+      description: Retrieve all the phases for a particular project with given ID.
+      tags:
+        - Phases
+      parameters:
+        - name: severaProjectGuid
+          in: path
+          required: true
+          description: ID of the project.
+          schema:
+            type: string
+      responses:
+        "200":
+          description: Phases entry retrieved successfully
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Phase"
+        "default":
+          description: Invalid request was sent to the server.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Error"
+
+  "/severa/{severaUserGuid}/resourceallocations":
+    get:
+      operationId: getAllocationsBySeveraUserGuid
+      summary: Get the resource allocations for a user.
+      description: Retrieve the resource allocations for a user based on their Severa GUID.
+      tags:
+        - ResourceAllocations
+      parameters:
+        - name: severaUserGuid
+          in: path
+          required: true
+          description: severaGuid of the user.
+          schema:
+            type: string
+      responses:
+        "200":
+          description: Resource allocations retrieved successfully
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/ResourceAllocations"
+        "default":
+          description: Invalid request was sent to the server.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Error"
+
+  "/severa/users/{severaUserGuid}/workhours":
+    get:
+      operationId: getWorkHoursBySeveraProjectId
```

Comment:
Is it the workhours for a project?  I see only the userId in use, how do we define which project? Or is the id and summary incorrectly named?

Reply:
Looking at the workhours schema I assume we can define the project/ phase the workhours refer to. I assume this is done using query params, in which case there query params should be defined here. Are some of them optional?

Reply:
Yes. All of them are optional, but we will add the lacking params

Reply:
Actually, they are not defined in the params. Project/phase is a property of workhours. So they should be stored in the same entity in the database, not defined manually in params. Let me know if I misunderstood your question.